3363 - Add admin and reviewers as recipients when sending the report for approval & Add notify self feature

yaguzmang opened the pull request:
Adding admin and reviewers as recipients when sending the report for approval, and adding notify-self feature.

TODO:
- Safely test email service. For this PR I tested locally that the email promises are correctly constructed, but didn't send actual emails. 


https://github.com/openforis/fra-platform/assets/41337901/f4ae763e-4ae7-4e99-b2f4-a1a31cb4ba74

yaguzmang committed:
Move useRecipients to hooks folder

yaguzmang committed:
Add notifySelf feature

yaguzmang committed:
Add reviewers and collaborator to approval roles

yaguzmang committed:
Sort collaborators list when updating status to approval

yaguzmang committed:
Add notify self button

yaguzmang committed:
Hide opt-out button for approval status

mergify[bot] committed:
Merge branch 'development' into 3363-add-admin-and-reviewers-as-recipients-when-sending-the-report-for-approval

minotogna commented:
Collaborators , Alternate NC and NC should not be recipient when sending the reports for approval (the issues states only admin and reviewers should). Is this the case in this pr?

mergify[bot] committed:
Merge branch 'development' into 3363-add-admin-and-reviewers-as-recipients-when-sending-the-report-for-approval

yaguzmang committed:
Remove collaborators from approval status

mergify[bot] committed:
Merge branch 'development' into 3363-add-admin-and-reviewers-as-recipients-when-sending-the-report-for-approval

mergify[bot] committed:
Merge branch 'development' into 3363-add-admin-and-reviewers-as-recipients-when-sending-the-report-for-approval

mergify[bot] committed:
Merge branch 'development' into 3363-add-admin-and-reviewers-as-recipients-when-sending-the-report-for-approval

minotogna commented:
@yaguzmang the pr looks good. Please, put it in draft and once you get the translations put it back to review and ask @sorja for his approval.
Thank you!

yaguzmang committed:
Add logic to send emails to admins

sorja commented on src/i18n/resources/en.js:
```diff
@@ -418,6 +418,7 @@ The FRA team
     cancel: 'Cancel',
     changeStatusTextPlaceholder: 'Add an optional message',
     doNotNotifyUsers: "Don't notify users",
+    notifySelf: 'Send copy to self',
```
Not sure, sounds weird
Maybe:

```suggestion
notifySelf: 'Send a copy to myself',
```

yaguzmang replied:
Would we want to requests translations again to keep it consistent? if so, it would take some time. We are still waiting for the French translation.

sorja replied:
I think it's fine to change or correct the English one if there is a typo?
All is fine, you can resolve this comment also without changing

yaguzmang committed:
Add missing translations

yaguzmang committed:
Fix wrong recipients shown

Fixing an issue that was causing users to be shown
as recipients if they had the required role in any
country, not just in the current country.

mergify[bot] committed:
Merge branch 'development' into 3363-add-admin-and-reviewers-as-recipients-when-sending-the-report-for-approval

mergify[bot] committed:
Merge branch 'development' into 3363-add-admin-and-reviewers-as-recipients-when-sending-the-report-for-approval

minotogna commented on src/server/repository/public/user/getAdmins.ts:
```diff
@@ -0,0 +1,34 @@
+      group by ${selectFields}
+  `
+
+  return client.manyOrNone<User>(query).then((data) =>
```
Here we usually use client.map . Also do we have UserAdapter directly ? If yes , use that , otherwise no worries we'll add it later

yaguzmang replied:
It seems we don't have a UserAdapter yet. For Users it seems we only have the UserRoleAdapter.

minotogna replied:
ok, but let's use client.map at least

yaguzmang replied:
Added to the latest commit ✌️

mergify[bot] committed:
Merge branch 'development' into 3363-add-admin-and-reviewers-as-recipients-when-sending-the-report-for-approval

yaguzmang committed:
Use client.map in getAdmins
